install/kubernetes/helm/istio/requirements.yaml

```diff
@@ -17,7 +17,6 @@ dependencies:
     repository: file://../subcharts/gateways
   - name: mixer
     version: 1.1.0
-    condition: mixer.policy.enabled,mixer.telemetry.enabled
```

Comment:
sorry - what issue this has caused?

Reply:
if you `--set mixer.policy.enabled=false --set mixer.telemetry.enabled=true`, you would expect to still get mixer config installs and the deployment for `istio-telemetry`. However, you instead get **no** mixer deployments.

Reply:
Ah thanks, I see it now in my env.  I agree the condition is problematic and not working well for multiple values.

Reply:
Updated to match the 1.1 PR that merged already.

Reply:
The new construct "condition: or mixer.policy.enabled mixer.telemetry.enabled" doesn't work for me.

I'm trying to install a custom Ingress Gateway. For that, I need to disable `mixer` subchart. However, it remains enabled despite `mixer.policy.enabled=false` and `mixer.telemetry.enabled=false`.

I don't see any support for "condition: or ..." in Helm sources (https://github.com/helm/helm/blob/master/pkg/chartutil/requirements.go#L127). 
Do I need some specific version of Helm for that construct to work?

Here is my Helm command:

```
helm template install/kubernetes/helm/istio \
  --name istio \
  --namespace istio-ingress-internal \
  --values install/kubernetes/helm/istio/values-istio-gateways.yaml \
  --set gateways.istio-egressgateway.enabled=false \
  --set gateways.istio-ingressgateway.enabled=false \
  --set gateways.custom-gateway.labels.istio=ingress-internal \
  --set mixer.enabled=false 
  --set mixer.policy.enabled=false  
  --set mixer.telemetry.enabled=false
```
The generated output still contains `mixer` parts.

Environment:
Istio: 1.1.0-snapshot.3
Helm: v2.12.1

Reply:
I think this may be related to using the `values-istio-gateways.yaml` values file.  Without that line, it works as expected.

Reply:
I tried without `values-istio-gateways.yaml`. 
But `mixer` components are still in the output.

The behaviour I'm observing is actually consistent with the source code of Helm (https://github.com/helm/helm/blob/master/pkg/chartutil/requirements.go#L127):
- Helm interprets `condition: or mixer.policy.enabled mixer.telemetry.enabled` as a single path `or mixer.policy.enabled mixer.telemetry.enabled`
- since path `or mixer.policy.enabled mixer.telemetry.enabled` doesn't exist, `mixer` plugin is always enabled 

So, I'm wondering what makes that syntax work for you.

Reply:
Also reproducible on `v2.11.0`.

But I noticed that my description of the problem wasn't complete.

With `mixer.policy.enabled=false` and `mixer.telemetry.enabled=false` the generated YAML:
- doesn't include:
-- istio/charts/mixer/templates/autoscale.yaml
-- istio/charts/mixer/templates/deployment.yaml
-- istio/charts/mixer/templates/service.yaml
- but still includes:
-- istio/charts/mixer/templates/serviceaccount.yaml
-- istio/charts/mixer/templates/clusterrole.yaml
-- istio/charts/mixer/templates/clusterrolebinding.yaml
-- istio/charts/mixer/templates/config.yaml

So, I was expecting that `mixer` subchart would be disabled based on conditions in `install/kubernetes/helm/istio/requirements.yaml`. But, apparently, it's not the case. Instead, `mixer` subchart is always enabled and some of its templates might render empty. 

As a user, I would expect a consistent behaviour - either a complete `mixer` configuration or none.